### PR TITLE
Fix coverity bugs - Week 9 and others

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#22183|
|#22184|

## Objective
This PR aims to address and resolve all suggestions highlighted by the Coverity analysis conducted in week 9, as part of our ongoing commitment to maintain high code quality and ensure robust security practices. 

## Description
- We have undertaken a comprehensive review of the Coverity report and implemented fixes for all identified issues. Each fix was carefully designed to not only address the immediate concern highlighted by Coverity but also to align with best coding practices and minimize the impact on existing functionalities.

### Static Analysis
- Post-fix, the code should be re-analyzed using Coverity to ensure that all previously identified issues were adequately addressed and resolved.

#### Unit Testing
N/A

#### Integration Testing
N/A

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp [hunk 1]
```diff
@@ -220,11 +220,11 @@ int main(const int argc, const char* argv[])
                         }
                         else if (tokens[0] == "agent" && Utils::isNumber(tokens[1]))
                         {
-                            auto agentId = tokens[1].length() < 3 ? "00" + tokens[1] : tokens[1];
+                            const auto& agentId = tokens[1].length() < 3 ? "00" + tokens[1] : tokens[1];
                             if (tableName.find("sys_programs") != std::string::npos || tokens[2] == "package")
                             {
                                 std::string responseMessage = fakeAgentPackages.contains(agentId)
-                                                                  ? "ok " + fakeAgentPackages[agentId].dump()
+                                                                  ? "ok " + fakeAgentPackages[std::move(agentId)].dump()
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp:227]: You can't move a const reference

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp]: Good catch!

File: src/wazuh_db/wdb_pool.c [hunk 2]
```diff
@@ -24,8 +24,8 @@ STATIC wdb_pool_t wdb_pool;
 
 __attribute__((no_thread_safety_analysis))
 void wdb_pool_init() {
-    wdb_pool.nodes = rbtree_init();
     w_mutex_init(&wdb_pool.mutex, NULL)
+    wdb_pool.nodes = rbtree_init();
```
Comment (sebasfalcone) [inline src/wazuh_db/wdb_pool.c:28]: We should acquire the mutex
```suggestion
    w_mutex_init(&wdb_pool.mutex, NULL)
    w_mutex_lock(&wdb_pool.mutex)
    wdb_pool.nodes = rbtree_init();
    w_mutex_unlock(&wdb_pool.mutex);
```

File: src/wazuh_db/wdb_pool.c
Comment (sebasfalcone) [inline src/wazuh_db/wdb_pool.c:1]: For the 
```C++
unsigned wdb_pool_size() {
    return wdb_pool.size;
}
```

Method we should also acquire the and release the mutex

**Review by sebasfalcone — COMMENTED** [review]
@GabrielEValenzuela 
Restore the `src/wazuh_db/wdb_pool.c` and `src/shared_modules/utils/socketDBWrapper.hpp` files, I think this should be done in another time and with the proper testing, as is not related to 4.8.0

**Review by sebasfalcone — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**